test/e2e_node/dra_test.go

```diff
@@ -0,0 +1,229 @@
+		})
+
+		ginkgo.It("must process pod created when kubelet is not running", func(ctx context.Context) {
+			// Stop Kubelet
+			startKubelet := stopKubelet()
+			pod := createTestObjects(ctx, f.ClientSet, getNodeName(ctx, f), f.Namespace.Name, "draclass", "external-claim", "drapod")
+			// Pod must be in pending state
+			err := e2epod.WaitForPodCondition(ctx, f.ClientSet, f.Namespace.Name, pod.Name, "Pending", podTimeout, func(pod *v1.Pod) (bool, error) {
+				return pod.Status.Phase == v1.PodPending, nil
+			})
+			framework.ExpectNoError(err)
+			// Start Kubelet
+			startKubelet()
+			// Pod should succeed
+			err = e2epod.WaitForPodSuccessInNamespaceTimeout(ctx, f.ClientSet, pod.Name, f.Namespace.Name, podTimeout)
+			framework.ExpectNoError(err)
+		})
+	})
+})
+
+// Run Kubelet plugin and wait until it's registered
+func newKubeletPlugin(nodeName string) *testdriver.ExamplePlugin {
+	ginkgo.By("start Kubelet plugin")
+	logger := klog.LoggerWithValues(klog.LoggerWithName(klog.Background(), "kubelet plugin"), "node", nodeName)
+
+	// Ensure that directories exist, creating them if necessary. We want
+	// to know early if there is a setup problem that would prevent
+	// creating those directories.
+	err := os.MkdirAll(cdiDir, os.FileMode(0750))
+	framework.ExpectNoError(err, "create CDI directory")
+	err = os.MkdirAll(filepath.Dir(endpoint), 0750)
+	framework.ExpectNoError(err, "create socket directory")
+
+	plugin, err := testdriver.StartPlugin(
+		logger,
+		cdiDir,
+		driverName,
+		"",
+		testdriver.FileOperations{},
+		kubeletplugin.PluginSocketPath(endpoint),
+		kubeletplugin.RegistrarSocketPath(path.Join(pluginRegistrationPath, driverName+"-reg.sock")),
+		kubeletplugin.KubeletPluginSocketPath(draAddress),
+	)
+	framework.ExpectNoError(err)
+
+	gomega.Eventually(plugin.IsRegistered).WithTimeout(time.Second * 30).Should(gomega.BeTrue())
+
+	ginkgo.DeferCleanup(plugin.Stop)
+
+	return plugin
+}
+
+// createTestObjects creates objects required by the test
+// NOTE: as scheduler and controller manager are not running by the Node e2e,
+// the objects must contain all required data to be processed correctly by the API server
+// and placed on the node without involving the scheduler and the DRA controller
+func createTestObjects(ctx context.Context, clientSet kubernetes.Interface, nodename, namespace, className, claimName, podName string) *v1.Pod {
+	// ResourceClass
+	class := &resourcev1alpha2.ResourceClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: className,
+		},
+		DriverName: driverName,
+	}
+	_, err := clientSet.ResourceV1alpha2().ResourceClasses().Create(ctx, class, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+
+	ginkgo.DeferCleanup(func() {
+		err := clientSet.ResourceV1alpha2().ResourceClasses().Delete(context.Background(), className, metav1.DeleteOptions{})
+		framework.ExpectNoError(err, "delete resource class")
+	})
```

Comment:
This can become a single line:

```suggestion
ginkgo.DeferCleanup(clientSet.ResourceV1alpha2().ResourceClasses().Delete, className, metav1.DeleteOptions{})
```

Gingko will do the error checking for us. We have to drop the annotation, but that's okay because the error is going to be fairly detailed.

Reply:
Same below.

Reply:
done, thank you.